### DISCUSSION TITLE
Influence of option --index-chunks on I/O

### DISCUSSION THREAD

**Discussion by duboism** [bbuchfink/diamond#648] [Q&A] [answered]
Hello,

We use `diamond` to match sequences against a rather large DB (around 80GB). In our case, we launch `diamond` independently on each sequence (I know it's not the most efficient way but this is rather hard to change in our software). We routinely process 10 sequences at the same time.

We are considering using `--index-chunks 1` as this seems to increase performance (currently we use the default value) but I was wondering if this would not cause many I/O on the same files given that several independent processes will access them (and thus slow down everything). More precisely, I am wondering if this will cause more I/O than the default value.

Any advice on this situation ?

Thanks in advance,
Mathieu

P.S.: Sorry if my question is dumb, I don't really know `diamond`.

**Comment by bbuchfink** [comment, marked as answer]
No, `--index-chunks 1` does not increase I/O load and should always be used if possible.

**Comment by duboism** [comment]
Thanks for you answer (your user support is great) !  I was running some tests that indicate exactly this. This will probably save us a lot of CPU time.
